src/css/texts.scss

```diff
@@ -11,104 +11,58 @@
   font-family: Roboto, sans-serif;
 }
 
-.text {
-  color: white;
-  font-weight: 400;
-  font-size: 30px;
-
-  @media #{$wide} {
-    font-size: 20px;
-  }
-
-  @media #{$regular} {
-    font-size: 16px;
-  }
-
-  @media #{$minimum} {
-    font-size: 14px;
-  }
-
-  @media #{$short} {
-    font-size: 12px;
-  }
-
-  @media #{$veryShortAndNarrow} {
-    font-size: 10px;
+/*
+ * Since we have a very weird scale of resolutions to support, from very narrow and tall
+ * to very "flat" and wide. Relative font sizes to the viewport broke down in these edge
+ * cases, hence there are 2 rules: body font size, which is the basis for everything and
+ * scales the font when width decreases; and main element font-size which scales the 
+ * font when height changes. The different em font sizes scale the component font sizes 
+ * relative to these two aforementioned baselines.
+ */
+
+body {
+  font-size: 22px;
+  @for $i from 1 through 5 {
+    @media all and (max-width: 1200px - $i * 200px) {
+      font-size: 22px - $i * 2px;
+    }
   }
 }
 
-.text--large {
+main {
```

Comment:
Now that the main is no longer the first element nested under `body`, maybe we should apply these rules to the `.main__container` instead?